extensions/smallrye-fault-tolerance/runtime/src/main/java/io/quarkus/smallrye/faulttolerance/runtime/QuarkusFaultToleranceOperationProvider.java

```diff
@@ -1,22 +1,39 @@
 package io.quarkus.smallrye.faulttolerance.runtime;
 
 import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 import javax.annotation.Priority;
 import javax.enterprise.context.Dependent;
 import javax.enterprise.inject.Alternative;
 
+import org.jboss.logging.Logger;
+
 import io.smallrye.faulttolerance.FaultToleranceOperationProvider;
 import io.smallrye.faulttolerance.config.FaultToleranceOperation;
 
 @Dependent
 @Alternative
 @Priority(1)
 public class QuarkusFaultToleranceOperationProvider implements FaultToleranceOperationProvider {
+    private static final Logger LOG = Logger.getLogger(QuarkusFaultToleranceOperationProvider.class);
+    static final Map<String, FaultToleranceOperation> OPERATIONS = new ConcurrentHashMap<>();
```

Comment:
You can make the `QuarkusFaultToleranceOperationProvider` a `@Singleton` and then there is no need to use static field here. Instead, you can use `Arc.container().instance(QuarkusFaultToleranceOperationProvider.class).init(Map<String,FaultToleranceOperation> ops)` in the recorder.

Reply:
OK, will do.
And if I don't implements `computeIfAbsent` I should be able to use an HashMap directly instead of a concurrent one as the map will never be modified outside the enhancer.

Reply:
Yep, an optimized immutable map would be a perfect fit but unfotunately there is no such impl in JDK.